### PR TITLE
Round 1 of QA Feedback

### PR DESCRIPTION
**Pull request by biyeun**
- fixes issue where deleting a column with filters doesn't refresh data
- removes ability to sort from ui (temporary for beta)
- updates how initial loading spinner is displayed
- prevents loading text from overlapping with table header

### CODE REVIEW

File: corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html [hunk 1]
```diff
@@ -78,11 +78,21 @@
       {% trans "Learn more." %}
     </a>
   </p>
-  <div class="ko-loading">
-    <i class="fa fa-spinner fa-spin"></i>
-    {% blocktrans %}
-      Loading...
-    {% endblocktrans %}
+
+  <div id="datagrid-initial-loading"
```
Comment (proteusvacuum) [inline corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html]: Should the class and ID of this node be named the same thing? I feel like that is slightly confusing.

File: corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js [hunk 2]
```diff
@@ -45,6 +45,11 @@ hqDefine('reports/v2/js/datagrid/data_models', [
                 .css('left', position.left + 'px')
                 .css('top', position.top + "px");
 
+            if (marginTop < 0) {
+                // This makes sure there's no overlap with the header.
+                marginTop = 0;
```
Comment (proteusvacuum) [inline corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js]: Couldn't you do 
``` js
marginTop = Math.min(0, $rows.height() / 2 - 50);
``` 
when you define `marginTop` above?

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html (hunk 1 above)
Comment (biyeun) [reply corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html]: done

File: corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js (hunk 2 above)
Comment (biyeun) [reply corehq/apps/reports/static/reports/v2/js/datagrid/data_models.js]: done

**Pull request merged**